File: Backend/SecureBooking.Domain/Entities/Room.cs
using System;
using SecureBooking.Shared.Enums;

namespace SecureBooking.Domain.Entities;

public class Room
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public string Name { get; set; } = string.Empty;

    public RoomType Type { get; set; }

    public string? Description { get; set; }

    public int Capacity { get; set; }

    public decimal PricePerNight { get; set; }

    public string? ImageUrl { get; set; }

    public bool IsActive { get; set; } = true;

    public Guid HotelId { get; set; }
    public Hotel? Hotel { get; set; }

    public ICollection<Booking> Bookings { get; set; } = [];
}

File: Frontend/secure-booking-web/src/types/Room.ts
export enum RoomType {
  Standard = "Standard",
  Deluxe = "Deluxe",
  Suite = "Suite",
  Family = "Family",
}

export interface Room {
  id: string;
  name: string;
  type: RoomType;
  description?: string;
  capacity: number;
  pricePerNight: number;
  imageUrl?: string;
  isActive: boolean;
  hotelId: string;
  hotelName: string;
  bookingCount: number;
  createdAt: string;
}
